src/soar_sdk/views/template_filters.py

```diff
@@ -0,0 +1,223 @@
+"""Jinja filters and functions for SOAR SDK templates. (ported and modified from platform custom_template.py)"""
+
+import hashlib
+import json
+import re
+import uuid
+import humanize
+import bleach  # type: ignore[import-untyped]
+from datetime import datetime, timedelta
+from typing import Optional, Union
+from collections.abc import Iterator
+from jinja2 import Environment
+
+
+def widget_uuid(length: int = 8) -> str:
+    """Generate a unique widget ID."""
+    unique_id = f"widget_{uuid.uuid4()}"
+    return unique_id.replace("-", "")
+
+
+def datetime_minutes(dt: timedelta) -> int:
+    """Convert datetime to minutes."""
+    return (dt.days * 24 * 60) + (dt.seconds // 60)
+
+
+def human_datetime(value: datetime, relative: bool = True) -> str:
+    """Format datetime in human readable format."""
+    if humanize and relative:
+        return humanize.naturaltime(value)
+    else:
+        return "{dt:%b} {dt:%-d}, {dt:%Y} {dt:%-I}:{dt:%M} {ampm}".format(
+            dt=value, ampm=value.strftime("%p").lower()
+        )
+
+
+def human_timedelta(td: timedelta) -> str:
+    """Format timedelta in human readable format."""
+    if humanize:
+        return humanize.naturaldelta(td)
+    return str(td)
+
+
+def sorteditems(dictionary: dict) -> list:
+    """Sort dictionary items by key."""
+    result = []
+    for key in sorted(dictionary):
+        result.append((key, dictionary[key]))
+    return result
+
+
+def batch(iterable: list, count: int) -> Iterator[list]:
```

Comment:
This can be generalized to any iterable:
```suggestion
def batch(iterable: Iterable, count: int) -> Iterator[list]:
```